### PR TITLE
[CosyVoice3] Vocoder acceleration.

### PR DESCRIPTION
**Pull request by PoTaTo-Mika**
## Motivation

Model pipeline haven't supported HiFT batch inference yet, and its CPU preprocessing and scheduler needs better refactor.

## Modifications

Only CosyVoice3 side code.

## Related Issues

#1652 #1655 #1693

## Accuracy Test

No accuracy decline.

## Checklist

- [x] Format your code according with pre-commit.
- [x] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

## CI

CI runs on self-hosted GPU runners and requires a maintainer to add the
`run-ci` label. Once labeled, every subsequent push re-triggers CI as
long as the label remains. Use `/tag-and-rerun-ci higgs` or
`/tag-and-rerun-ci moss` or `/tag-and-rerun-ci qwen3-tts` to select a TTS CI
model, and
`/tag-and-rerun-ci fun-asr`, `/tag-and-rerun-ci qwen3-asr` or
`/tag-and-rerun-ci whisper-asr` to select an ASR CI model. One selector from
each family can be combined, for example `/tag-and-rerun-ci moss fun-asr`.
Draft PRs are skipped even if labeled.

### CODE REVIEW

File: sglang_omni/models/fun_cosyvoice3/stages.py [hunk 1]
```diff
@@ -585,9 +706,43 @@ def _flow_scheduler_cost(self, payload: StagePayload) -> int:
...
+        if len(mels) == 1:
+            return [self._mel2wav(mels[0])]
+        lengths = [int(mel.shape[2]) for mel in mels]
+        longest = max(lengths)
+        if min(lengths) == longest:
+            padded = torch.cat(mels, dim=0)
+        else:
+            padded = torch.cat(
+                [
+                    F.pad(mel, (0, longest - length))
```
Comment (AkazaAkane) [inline sglang_omni/models/fun_cosyvoice3/stages.py:709]: Could we either:

1. keep HiFT batching restricted to equal-length mels as #1655 or
2. provide real-checkpoint serial-vs-padded validation, especially around the waveform tail/end boundary, showing the numerical/audio-quality delta is acceptable?

I think padding variable-length mels here changes HiFT semantics, so this needs stronger correctness validation before we enable it. CosyVoice3 HiFT calls torch.stft on the generated source with the default centered STFT behavior. For a standalone shorter utterance, its real endpoint is an STFT boundary. After right-padding it to another request's length, that endpoint becomes an interior point followed by artificial zeros. Cropping the final waveform back to length * samples_per_frame does not necessarily recover the serial result because computation near the original boundary has already seen different context.

File: sglang_omni/models/fun_cosyvoice3/config.py [hunk 2]
```diff
@@ -55,7 +60,9 @@ def process_local_edges(cls) -> frozenset[tuple[str, str]]:
             factory=FactoryArgs(
                 dtype="bfloat16",
                 flow_batch_bucket_frames=50,
-                flow_batch_admission_frames=2000,
+                flow_batch_admission_frames=8000,
+                max_batch_size=16,
+                max_batch_wait_ms=30,
```
Comment (SandyLuXY) [inline sglang_omni/models/fun_cosyvoice3/config.py:65]: Can we add the benchmarking results in the PR body to validate the perf?

File: sglang_omni/models/fun_cosyvoice3/config.py [hunk 3]
```diff
@@ -55,7 +60,9 @@ def process_local_edges(cls) -> frozenset[tuple[str, str]]:
             factory=FactoryArgs(
                 dtype="bfloat16",
                 flow_batch_bucket_frames=50,
-                flow_batch_admission_frames=2000,
+                flow_batch_admission_frames=8000,
```
Comment (SandyLuXY) [inline sglang_omni/models/fun_cosyvoice3/config.py:63]: need to update cookbook `docs/cookbook/fun_cosyvoice3.md` for the changes

**Comment by PoTaTo-Mika** [comment]
## English (en, 1088 samples)

| Concurrency | Success | Failed | QPS (req/s) | Audio throughput (s/s) | Latency mean (s) | Latency median (s) | Latency P95 (s) | Latency P99 (s) | RTF mean | RTF P95 | Audio duration mean (s) | Output tokens/req | Output tokens/s |
|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
| 16 | 1088 | 0 | 7.545 | 38.529 | 2.116 | 2.022 | 2.876 | 3.382 | 0.4653 | 0.7463 | 5.107 | 128 | 963.2 |

## Chinese (zh, 2020 samples)

| Concurrency | Success | Failed | QPS (req/s) | Audio throughput (s/s) | Latency mean (s) | Latency median (s) | Latency P95 (s) | Latency P99 (s) | RTF mean | RTF P95 | Audio duration mean (s) | Output tokens/req | Output tokens/s |
|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
| 16 | 2020 | 0 | 7.406 | 48.401 | 2.156 | 2.133 | 2.941 | 3.519 | 0.3394 | 0.4638 | 6.535 | 163 | 1210.0 |

**Review by Dayuxiaoshui — APPROVED** [review]
